### PR TITLE
fix(Select): Fix select options positioning inside non static parents

### PR DESCRIPTION
**Pull request by pladaria**
Fixes this: https://t.ly/Yb_W (open select)

### CODE REVIEW

File: src/select.tsx [hunk 1]
```diff
@@ -106,12 +106,19 @@ const Select: React.FC<SelectProps> = ({
...
-                    height,
-                } = fieldRef.current.getBoundingClientRect();
+
+                // get a possible non-static parent, like a modal dialog
+                const nonStaticParentElement = focusableRef.current?.offsetParent;
+                const parentElementRect = nonStaticParentElement?.getBoundingClientRect() || {
+                    left: 0,
+                    top: 0,
+                };
+                const clientRect = fieldRef.current.getBoundingClientRect();
```
Comment (marcoskolodny) [inline src/select.tsx:116]: Calling this function may cause errors if using server side rendering, right?

Comment (pladaria) [reply src/select.tsx:116]: yes, but this is only executed when interacting with the element

**Review by atabel — APPROVED** [review]

**Review by marcoskolodny — APPROVED** [review]

**Review by yceballost — APPROVED** [review]
perfectou!

**Pull request merged**